From 214eba2e7a9c080223a172dbd6a9c6f74e2f5b3f Mon Sep 17 00:00:00 2001
From: Kel Modderman <kelvmod@gmail.com>
Date: Sun, 9 Aug 2026 18:27:20 +1000
Subject: [PATCH] Read the plist from stdin without an explicit -

xxd reads standard input when given no file operand, so the - is
redundant. It also breaks builds that provide xxd through OpenWrt's
scripts/xxdi.pl, a partial reimplementation which treats - as a file
name and exits with

  xxdi.pl: Unable to open -: No such file or directory

The script does not check for that failure, so the generated
get_info_response.c ends up with an empty byte array and AirPlay 2
answers GET /info with nothing.

Upstream-Status: Backport [https://github.com/mikebrady/shairport-sync/commit/214eba2e7a9c080223a172dbd6a9c6f74e2f5b3f]

---

--- a/xml_plist_codegen.sh
+++ b/xml_plist_codegen.sh
@@ -23,6 +23,6 @@ printf "// Please do not edit this file!
 printf "// This file was automatically generated from $INPUT_FILE_NAME.\n\n" >> $2/$BASE_FILENAME.c
 printf "#include \"$BASE_FILENAME.h\"\n\n" >> $2/$BASE_FILENAME.c
 printf "unsigned char ${BASE_FILENAME}_plist[] = {\n" >> $2/$BASE_FILENAME.c
-plistutil -i $1 | xxd -i - >> $2/$BASE_FILENAME.c
+plistutil -i $1 | xxd -i >> $2/$BASE_FILENAME.c
 printf "  };\n\n" >> $2/$BASE_FILENAME.c
 printf "size_t ${BASE_FILENAME}_plist_len = sizeof(${BASE_FILENAME}_plist);\n" >> $2/$BASE_FILENAME.c
